main_test.go

```diff
@@ -29,7 +29,7 @@ func TestRunErrors(t *testing.T) {
 	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
 		w.WriteHeader(200)
 	}))
-	defer server.Close()
+	t.Cleanup(func() { server.Close() })
```

Comment:
let's not cruft-sandwich things. everywhere I'll ask to not func() wrap functions

```suggestion
	t.Cleanup(server.Close)
```